lib/rules/declaration-property-max-values/index.js

```diff
@@ -0,0 +1,56 @@
+'use strict';
+
+const matchesStringOrRegExp = require('../../utils/matchesStringOrRegExp');
+const report = require('../../utils/report');
+const ruleMessages = require('../../utils/ruleMessages');
+const valueParser = require('postcss-value-parser');
+const vendor = require('../../utils/vendor');
+
+const ruleName = 'declaration-property-max-values';
+
+const messages = ruleMessages(ruleName, {
+	rejected: (property, max) =>
+		`Expected "${property}" to have no more than ${max} ${max === 1 ? 'value' : 'values'}`,
+});
+
+const meta = {
+	url: 'https://stylelint.io/user-guide/rules/list/declaration-property-max-values',
+};
+
+/** @type {import('stylelint').Rule<Record<string, Number>>} */
+const rule = (primary) => {
+	return (root, result) => {
```

Comment:
[suggest] I think `validateOptions` is necessary in almost all cases.

```js
const validOptions = validateOptions(...);
```

Reply:
Should we just use `isPlainObject` introduced in https://github.com/stylelint/stylelint/pull/5922?

Or should be more specific, e.g. this rule expects an object whose values are numbers?

Reply:
I think `validateObjectWithArrayProps` may be helpful in this case. E.g.

```js
const validOptions = validateOptions(result, ruleName, {
	actual: primary,
	possible: [validateObjectWithArrayProps(isNumber)],
});
```

https://github.com/stylelint/stylelint/blob/f1ee719fe537bfa41b248df1bc7ae5528c153fff/lib/utils/validateObjectWithArrayProps.js#L22

Reply:
As this rule expects a single number (rather than an array of them), we may need a new function similar to `validateObjectWithArrayProps`.

Shall we create `validateObjectWith()`, e.g.:

```js
const validOptions = validateOptions(result, ruleName, {
	actual: primary,
	possible: [validateObjectWith(isNumber)],
});
```

Reply:
Nice idea! 👍🏼